pkg/config/database.go

```diff
@@ -75,7 +81,46 @@ func (d *Database) Open(logger *logging.Logger) (*icingadb.DB, error) {
 			}
 		}
 
-		dsn = config.FormatDSN()
+		c, err := mysql.NewConnector(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "can't open mysql database")
+		}
+
+		wsrepSyncWait := int64(d.Options.WsrepSyncWait)
+
+		// Setting "wsrep_sync_wait" for each session ensures that causality checks are performed before execution
+		// and that each statement is executed on a fully synchronized node. Doing so prevents foreign key violation
+		// when inserting into dependent tables on different MariaDB/MySQL nodes. When using MySQL single nodes, the
+		// "SET SESSION" command will fail with "Unknown system variable (1193)" and will therefore be silently dropped.
+		// https://mariadb.com/kb/en/galera-cluster-system-variables/#wsrep_sync_wait
+		var setGaleraOpts = func(ctx context.Context, conn driver.Conn) error {
```

Comment:
I don't understand why `setGaleraOpts` is set here and `errUnknownSysVar` on the package level. Either make both usable, or define both here.

Reply:
I believe that's a result of my suggestion to capture in the config option in https://github.com/Icinga/icingadb/pull/711#discussion_r1531695737.

But that could also be achieved like this:

```go
connector := &icingadbDriver.RetryConnector{
	Connector: c,
	SqlDriver: &mysql.MySQLDriver{},
	Logger: logger,
	InitConn: func(ctx context.Context, conn driver.Conn) error {
		return setWsrepSyncWait(ctx, conn, wsrepSyncWait)
	}
}
```

That would probably make the code a bit nicer to read compared to the large anonymous function currently in place.